renku/core/dataset/dataset.py

```diff
@@ -735,6 +735,9 @@ def show_dataset(name):
         dict: JSON dictionary of dataset details.
     """
     dataset = DatasetsProvenance().get_by_name(name)
```

Comment:
`DatasetProvenance::get_by_name` has a `strict` parameter which when set to `True` raises an `errors.DatasetNotFound` when dataset is not found. It's more consistent to use that parameter instead of repeating the logic here.